journal/src/main/scala/com/evolutiongaming/kafka/journal/PartitionCache.scala

```diff
@@ -576,22 +576,22 @@ private[journal] object PartitionCache {
      *
      * Same as [[Later.Behind]], but returns [[Result]]
      */
-    def behind[F[_]](value: F[Now]): Result[F] = Later.behind(value)
+    def behind[F[_]](value: F[Now[F]]): Result[F] = Later.behind(value)
 
     /**
      * The cache was empty when [[PartitionCache#get]] got called.
      *
      * Same as [[Later.Empty]], but returns [[Result]]
      */
-    def empty[F[_]](value: F[Now]): Result[F] = Later.empty(value)
+    def empty[F[_]](value: F[Now[F]]): Result[F] = Later.empty(value)
 
     /**
      * [[PartitionCache]] already seen such [[Offset]] in Kafka or Cassandra.
      *
      * In other words, it means that this offset was either already replicated, or had the latest
      * [[HeadInfo]] information inside of [[PartitionCache]] when [[PartitionCache#get]] was called.
      */
-    sealed trait Now extends Result[Nothing]
+    sealed trait Now[F[_]] extends Result[F]
```

Comment:
why do we need the type-class here?

Reply:
Because Result requires it, you can't extend it without a typeclass